src/Event/Http/HttpRequestEvent.php

```diff
@@ -125,25 +138,40 @@ public function getRequestContext(): array
 
     public function getCookies(): array
     {
-        if (! isset($this->headers['cookie'])) {
-            return [];
-        }
+        $cookies = [];
+        if ($this->getPayloadVersion() >= 2 && isset($this->event['cookies'])) {
```

Comment:
Here is a suggestion:

```suggestion
        if ($this->payloadVersion >= 2) {
```

Reasons:

- avoid calling the getter internally, very small perf optimization (I don't like doing those, but in these classes in Bref this makes sense)
- if we are in v2, we should not go in the `v1` code branch

Reply:
Cool, just making those changes now.